### PR TITLE
fix: agent default config override template

### PR DESCRIPTION
**Pull request by naseemkullah**
move `exporters` and `service.pipelines` into if statement to avoid
setting them to null when standalone is disabled

Signed-off-by: naseemkullah <naseem@transit.app>

### CODE REVIEW

**Comment by naseemkullah** [comment]
@tigrannajaryan PTAL 🙏

**Comment by tigrannajaryan** [comment]
@naseemkullah can you please clarify what problem is this solving?

File: charts/opentelemetry-collector/templates/_config.tpl [hunk 1]
```diff
@@ -105,19 +105,19 @@ Get otel memory_limiter ballast_size_mib value based on 40% of resources.memory.
 Default config override for agent collector deamonset
 */}}
 {{- define "opentelemetry-collector.agentConfigOverride" -}}
+{{- if .Values.standaloneCollector.enabled }}
 exporters:
```
Comment (tigrannajaryan) [inline charts/opentelemetry-collector/templates/_config.tpl:109]: `exporters` is a mandatory section for Collector config. Will this result in no `exporters` at all?

File: charts/opentelemetry-collector/templates/_config.tpl [hunk 2]
```diff
@@ -105,19 +105,19 @@ Get otel memory_limiter ballast_size_mib value based on 40% of resources.memory.
...
-  {{- if .Values.standaloneCollector.enabled }}
   otlp:
     endpoint: {{ include "opentelemetry-collector.fullname" . }}:55680
     insecure: true
-  {{- end }}
+{{- end }}
 
+{{- if .Values.standaloneCollector.enabled }}
```
Comment (tigrannajaryan) [inline charts/opentelemetry-collector/templates/_config.tpl:115]: Similarly, `service` is a mandatory config section.

**Review by tigrannajaryan — COMMENTED** [review]
I am not fully familiar with how the helm chart works, I left a couple comments.

File: charts/opentelemetry-collector/templates/_config.tpl (hunk 1 above)
Comment (naseemkullah) [reply charts/opentelemetry-collector/templates/_config.tpl:109]: no, however previous to this change, if standaloneCollector were disabled, `exporters`  would always be `null`

I will provide some example templated manifests before and after this change in the near future

File: charts/opentelemetry-collector/templates/_config.tpl (hunk 2 above)
Comment (naseemkullah) [reply charts/opentelemetry-collector/templates/_config.tpl:115]: as above, except for `service.piplines` instead of `exporters`

**Comment by naseemkullah** [comment]
> I am not fully familiar with how the helm chart works, I left a couple comments.

Thanks I will provide example templated manifests to show before and after which will provide a clearer understanding to what this fixes, but tl;dr is that  the agent's config's `exporters` and `service.pipelines` were being set to null by mistake when standalone collector is disabled.

**Comment by naseemkullah** [comment]
Hi @tigrannajaryan  

tl;dr this PR fixes the case where standalone is disabled, please note the values for `exporters` and `service.pipelines`

Here are the various configs generated by the chart:

1) in its current state, with default values (standalone disabled): 👎 

```yaml
    exporters: null
    extensions:
      health_check: {}
    processors:
      batch: {}
      memory_limiter: null
      queued_retry: {}
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250
          thrift_http:
            endpoint: 0.0.0.0:14268
      otlp:
        protocols:
          grpc: null
          http: null
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${MY_POD_IP}:8888
      zipkin:
        endpoint: 0.0.0.0:9411
    service:
      extensions:
      - health_check
      pipelines: null
```
 
2) in its current state, with standalone enabled: 👍 

```yaml
    exporters:
      logging: {}
      otlp:
        endpoint: RELEASE-NAME-opentelemetry-collector:55680
        insecure: true
    extensions:
      health_check: {}
    processors:
      batch: {}
      memory_limiter: null
      queued_retry: {}
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250
          thrift_http:
            endpoint: 0.0.0.0:14268
      otlp:
        protocols:
          grpc: null
          http: null
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${MY_POD_IP}:8888
      zipkin:
        endpoint: 0.0.0.0:9411
    service:
      extensions:
      - health_check
      pipelines:
        metrics:
          exporters:
          - otlp
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - prometheus
        traces:
          exporters:
          - otlp
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - jaeger
          - zipkin
```

3) with this PR, with default values (standalone disabled): 👍 

```yaml
    exporters:
      logging: {}
    extensions:
      health_check: {}
    processors:
      batch: {}
      memory_limiter: null
      queued_retry: {}
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250
          thrift_http:
            endpoint: 0.0.0.0:14268
      otlp:
        protocols:
          grpc: null
          http: null
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${MY_POD_IP}:8888
      zipkin:
        endpoint: 0.0.0.0:9411
    service:
      extensions:
      - health_check
      pipelines:
        metrics:
          exporters:
          - logging
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - prometheus
        traces:
          exporters:
          - logging
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - jaeger
          - zipkin
```
 
4) with this PR, with standalone enabled: 👍 

```yaml
    exporters:
      logging: {}
      otlp:
        endpoint: RELEASE-NAME-opentelemetry-collector:55680
        insecure: true
    extensions:
      health_check: {}
    processors:
      batch: {}
      memory_limiter: null
      queued_retry: {}
    receivers:
      jaeger:
        protocols:
          grpc:
            endpoint: 0.0.0.0:14250
          thrift_http:
            endpoint: 0.0.0.0:14268
      otlp:
        protocols:
          grpc: null
          http: null
      prometheus:
        config:
          scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
            - targets:
              - ${MY_POD_IP}:8888
      zipkin:
        endpoint: 0.0.0.0:9411
    service:
      extensions:
      - health_check
      pipelines:
        metrics:
          exporters:
          - otlp
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - prometheus
        traces:
          exporters:
          - otlp
          processors:
          - memory_limiter
          - batch
          - queued_retry
          receivers:
          - otlp
          - jaeger
          - zipkin
```

**Review by tigrannajaryan — APPROVED** [review]
Thanks @naseemkullah
